dashboard/config/initializers/devise.rb

```diff
@@ -315,6 +315,20 @@
     manager.failure_app = CustomDeviseFailure
   end
 
+  Warden::Manager.after_authentication do |user, auth|
+    if auth.cookies[:pm] == "new_header"
+      cookie_key = '_user_type' + (Rails.env.production? ? '' : "_#{Rails.env}")
```

Comment:
Yeah, I see that we would want to use environment-specific cookies for testing, since these are top-level domain cookies being set. I'll need to make the corresponding change in the http cache config PR but this sounds good.

Any reason for the leading underscore (`_`) in the cookie name? Just to mark that this cookie is meant for internal use?